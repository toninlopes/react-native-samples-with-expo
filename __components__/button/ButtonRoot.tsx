import { StyleSheet, TouchableOpacity } from "react-native";
import type { ButtonRootProps } from './types';

const ButtonRoot: React.FC<ButtonRootProps> = ({ children, disabled, ...rest }) => (
  <TouchableOpacity style={styles(disabled).button} disabled={disabled} {...rest}>
    {children}
  </TouchableOpacity>
);

const styles = (
  disabled?: boolean,
) => ({
  ...StyleSheet.create({
    button: {
      flexDirection: 'row',
      backgroundColor: 'silver',
      padding: 6,
      justifyContent: 'center',
      alignItems: 'center',
      borderRadius: 8,
      height: 42,
      opacity: disabled ? 0.4 : 1,
    },
    buttonText: {
      fontSize: 18,
    }
  }),
});

export default ButtonRoot;