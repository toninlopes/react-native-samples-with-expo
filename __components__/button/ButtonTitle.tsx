import { StyleSheet, Text } from "react-native";
import type { ButtonTitleProps } from './types';

const ButtonTitle: React.FC<ButtonTitleProps> = ({ children, ...rest }) => (
  <Text style={styles.text} {...rest}>{children}</Text>
);

const styles = StyleSheet.create({
  text: {
    fontSize: 18,
    alignSelf: 'center',
  },
});

export default ButtonTitle;