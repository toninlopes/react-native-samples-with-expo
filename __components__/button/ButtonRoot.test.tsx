import React from 'react';
import renderer from 'react-test-renderer';

import ButtonRoot from './ButtonRoot';
import ButtonTitle from './ButtonTitle'

describe('ButtonRoot', () => {
  test('When there is button title.', () => {
    const button = renderer.create((
      <ButtonRoot>
        <ButtonTitle>{'Title'}</ButtonTitle>
      </ButtonRoot>
    )).toJSON();
    expect(button).toMatchSnapshot();
  });
});