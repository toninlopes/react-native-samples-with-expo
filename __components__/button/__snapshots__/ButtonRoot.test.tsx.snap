// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonRoot When there is button title. 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "silver",
      "borderRadius": 8,
      "flexDirection": "row",
      "height": 42,
      "justifyContent": "center",
      "opacity": 1,
      "padding": 6,
    }
  }
>
  <Text
    style={
      {
        "alignSelf": "center",
        "fontSize": 18,
      }
    }
  >
    Title
  </Text>
</View>
`;
